### PR TITLE
Configuring a separate user for harp manager

### PR DESCRIPTION
**Pull request by gonzalemario**

### CODE REVIEW

File: roles/harp/config/tasks/harp2.yml [hunk 1]
```diff
@@ -103,20 +103,43 @@
...
+    register: config_yml
+    ignore_errors: true
+
+  - name: Check if we have set harp_manager_user already
+    assert:
+      that:
+        - "'{{ user_to_check }}' in '{{ item }}'"
+      fail_msg: >
+        We cannot change the manager user on existing clusters, please set
+        "harp_user_manager" in config.yml to use current user defined in "{{ item }}"
```
[the remark this answers is not in the document]
Comment (amenonsen) [reply roles/harp/config/tasks/harp2.yml:120]: But if you set harp_manager_user to x and deploy, and set it to y and redeploy, we don't care, we can change the user and restart harp. The point is to make the change opt-in (i.e., by setting harp_manager_user explicitly if required), and not to change old clusters unilaterally.

File: roles/init/tasks/postgres.yml [hunk 2]
```diff
@@ -228,6 +228,28 @@
...
+- set_fact:
+    current_harp_manager_user: "{{ item.split('user=')[1] | trim }}"
+
+  vars:
+    harp_config_struct: "{{ config_yml.stdout | from_yaml }}"
+    endpoints: "{{ harp_config_struct.dcs.endpoints }}"
+  loop: "{{ endpoints }}"
+  when: config_yml.rc == 0
+
+- set_fact:
```
Comment (wind39) [inline roles/init/tasks/postgres.yml:250]: In this case, IIUC from the code, `harp_manager_user` will assume:

1- The value the user specified in `harp_manager_user` from `config.yml`; OR

2- The current user from the HARP Manager DSN, which in previous deployed clusters would be `postgres`; OR

3- The default specified by `default_harp_manager_user`, which is `superhman`.

I agree with the behavior.

**Review by wind39 — DISMISSED** [review]
Looks correct to me.

File: roles/init/tasks/postgres.yml [hunk 3]
```diff
@@ -228,6 +228,28 @@
...
+# not possible because of reasons, a default will be given.
+
+- name: Read current configuration file if exists
+  shell: "cat {{ harp_directory }}/config.yml"
+  register: config_yml
+  ignore_errors: true
+
+- set_fact:
+    current_harp_manager_user: "{{ item.split('user=')[1] | trim }}"
+
```
Comment (haroon-github) [inline roles/init/tasks/postgres.yml:243]:
```suggestion
```

File: roles/init/defaults/main.yml [hunk 4]
```diff
@@ -186,6 +186,7 @@ default_harp_consensus_protocols:
 default_harp_directories:
   "2": /etc/harp
 default_harp_proxy_mode: builtin
+default_harp_manager_user: superhman
```
Comment (haroon-github) [inline roles/init/defaults/main.yml:189]:
```suggestion
default_harp_manager_user: harpmanager
```

**Review by haroon-github — APPROVED** [review]

**Pull request merged**